src/models/ServerStyleSheet.js

```diff
@@ -39,13 +39,25 @@ class ServerTag implements Tag {
   }
 
   toHTML() {
-    const namesAttr = `${SC_ATTR}="${this.names.join(' ')}"`
-    const localAttr = `${LOCAL_ATTR}="${this.isLocal ? 'true' : 'false'}"`
-    const css = Object.keys(this.components)
-      .map(key => this.components[key].css)
-      .join('')
+    const attrs = [
+      'type="text/css"',
+      `${SC_ATTR}="${this.names.join(' ')}"`,
+      `${LOCAL_ATTR}="${this.isLocal ? 'true' : 'false'}"`,
+    ]
+
+    /* eslint-disable no-underscore-dangle */
+    if (global.__webpack_nonce__) {
+      attrs.push(`nonce="${global.__webpack_nonce__}"`)
+    }
+    /* eslint-enable */
+
+    /* eslint-disable */
+    const css = Object.keys(this.components).reduce((styles, key) => {
+      return (styles += this.components[key].css), styles
```

Comment:
I actually ended up making this into a reusable function since the same thing happens in `toReactElement`